### PR TITLE
[Merged by Bors] - feat: port LinearAlgebra.Dimension

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

Rest assured that at the time of posting this PR, mathport had already incorporated the latest changes to this file. Cf. [Zulip](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/!4.233247).

* [`linear_algebra.dimension`@`3a2039ad20626aebbece10cddb18ef010f7d912d`..`45ce3929e3bf9a086a216feea3b1ab6c14bf0e67`](https://leanprover-community.github.io/mathlib-port-status/file/linear_algebra/dimension?range=3a2039ad20626aebbece10cddb18ef010f7d912d..45ce3929e3bf9a086a216feea3b1ab6c14bf0e67)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
> Rest assured that at the time of posting this PR, mathport had already incorporated the latest changes to this file. Cf. [Zulip](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/!4.233247).

The dashboard [says otherwise](https://leanprover-community.github.io/mathlib-port-status/file/linear_algebra/dimension)

**Comment by Parcly-Taxel** [comment]
> The dashboard says otherwise

Ported.

**Comment by Komyyy** [comment]
The only thing to do is to erase `maxHeartbeats` at `rank_add_rank_split`.

File: Mathlib/LinearAlgebra/Dimension.lean [hunk 1]
```diff
@@ -0,0 +1,1449 @@
...
+/-- If `R` satisfies the rank condition,
+then the cardinality of any basis is bounded by the cardinality of any spanning set.
+-/
+theorem Basis.le_span {J : Set M} (v : Basis ι R M) (hJ : span R J = ⊤) : (#range v) ≤ (#J) := by
+  haveI := nontrivial_of_invariantBasisNumber R
+  cases fintypeOrInfinite J
+  · rw [← Cardinal.lift_le, Cardinal.mk_range_eq_of_injective v.injective, Cardinal.mk_fintype J]
+    convert Cardinal.lift_le.{w, v}.2 (basis_le_span' v hJ)
+    simp
+  · -- Porting note: `have` in this line is unnecessary.
```
Comment (semorrison) [inline Mathlib/LinearAlgebra/Dimension.lean:643]: I'm happy if small emendations like this go through without a porting note. We don't need someone to review this after the port is complete.

File: Mathlib/LinearAlgebra/Dimension.lean [hunk 2]
```diff
@@ -0,0 +1,1449 @@
...
+section
+
+variable [AddCommGroup V₂] [Module K V₂]
+
+variable [AddCommGroup V₃] [Module K V₃]
+
+open LinearMap
+
+#eval "TODO: Erase this."
+set_option maxHeartbeats 270000 in
```
Comment (semorrison) [inline Mathlib/LinearAlgebra/Dimension.lean:1139]: Cleanup here?

**Comment by semorrison** [comment]
Can someone confirm that every one of the etaExperiments is actually needed? I'm struggling with getting oleans right now.

**Comment by semorrison** [comment]
Okay, looks good to me! Thanks all for the rapid progress on this one.

bors merge

**Comment by Komyyy** [comment]
All 36 decls require `etaExperiment`s.

**Comment by semorrison** [comment]
Ah, maybe we should look more at this maxHeartbeats thing.

**Comment by semorrison** [comment]
bors merge

**Comment by Komyyy** [comment]
Please wait, I removed `maxHeartbeats`.

**Comment by Parcly-Taxel** [comment]
> Please wait, I removed `maxHeartbeats`.

Well I'd love to see that; I just pushed a commit that neatens the formatting of the proofs here.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**